### ISSUE TITLE
[Bug] Find in page UI Test broken on API 25 for both fenix and r-b

### ISSUE THREAD

**Issue by rpappalax** [mozilla-mobile/fenix#5534]
## Steps to reproduce
Run findInPage automated test.
Pulls up a static html asset with known content.
searches for a specific combination of chars

https://github.com/mozilla-mobile/fenix/blob/master/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt#L145

### Expected behavior
Fenix should display 1/3 for value of find_in_page_result_text

### Actual behavior
Fenix displays 1/5 for value of find_in_page_result_text and thus fails on this line:https://github.com/mozilla-mobile/fenix/blob/master/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt#L145

### Device information

* Android device: Nexus9 (x86 devicer)
* Fenix version: master

**Comment by rpappalax** [comment]
<img width="460" alt="image" src="https://user-images.githubusercontent.com/3699494/65790587-cc9dcd00-e114-11e9-92f8-e67f83036e6b.png">

**Comment by rpappalax** [comment]
@sblatz ^ this is what we're seeing intermittently.  The find in page for an asset that has consistent static content should be 1/7 and is instead 11/42

**Comment by ekager** [comment]
It looks like there are maybe 42 "i"s on the page and it's updating incorrectly? This could be a bug with find in page that would have to be fixed in AC

**Comment by rpappalax** [comment]
@ekager yes, I'm seeing the same issue in r-b except in that case, the expect text is missing altogether

```
failed: findInPageTest(org.mozilla.reference.browser.ui.ThreeDotMenuTest)
09-26 17:42:56.222: E/TestRunner(11929): ----- begin exception -----
09-26 17:42:56.223: E/TestRunner(11929): androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'with text: is "1/3"' doesn't match the selected view.
09-26 17:42:56.223: E/TestRunner(11929): Expected: with text: is "1/3"
09-26 17:42:56.223: E/TestRunner(11929):      Got: "AppCompatTextView{id=2131296394, res-name=find_in_page_result_text, visibility=VISIBLE, width=0, height=43, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@bd975d8, tag=null, root-is-layout-requested=false, has-input-connection=false, x=660.0, y=52.0, text=, input-type=0, ime-target=false, has-links=false}"
```

**Comment by csadilek** [comment]
@rpappalax @ekager Hm, I can't reproduce this. I tried both manually as well as running `NavigationToolbarTest.findInPageTest` on various devices incl. API 25.

My guess is that the UI updates are slower on some CI builds. It looks like the test doesn't wait until the counter updates. It only waits until the counter is visible (which it is but still displaying the previous result). So, I think, the test needs an update, as the verification can happen before the UI is updated. We could naively wait until we verify or check for text changes and then verify.

This could've happened before the recent refactoring too but it makes sense this is more likely now as there is an additional thread involved.

**Comment by rpappalax** [comment]
@csadilek I can't repro this locally either, but it's happening in Firebase devices on API 25 for both R-B and fenix.  In the case of R-B, the text is absent.  In the case of fenix, the text is present but the count is extremely high in addition to being incorrect.  If you feel that this may be a corner case or non-issue, then I can turn off the test for API 25.  Otherwise, let me know if you'd recommend I file this as an issue under A-C.

**Comment by csadilek** [comment]
@rpappalax I think we need to update the test. It assumes that the UI (the result counter) updates immediately when the text is entered, but this can happen after a slight delay (there's a thread context switch involved). See my comment above for possible solutions. Let me know if you want to pair up on trying this.

**Comment by ekager** [comment]
We updated this test so let's see if there are future failures. Closing for now

**Comment by ekager** [comment]
Ugh reopening I'm still seeing this test failing on PRs intermittently

**Comment by ekager** [comment]
Unassigning myself for now because I'm not sure what else to try here

**Comment by sblatz** [comment]
@rpappalax should we temporarily disable this test? 😕

**Comment by csadilek** [comment]
:(. I will assign to me to see if I can reproduce somehow or find out more. Likely not this week, but I'll look into it.

**Comment by csadilek** [comment]
@ekager @rpappalax The test itself looks good to me with the new changes to wait for the UI updates! I can not get the tests to fail even when running it in a loop on multiple devices.

However, after looking into this and testing manually, I did run into issues on some sites, which would explain the intermittent failures we've been seeing. I've filed them in A-C for now (could also need GV fixes): https://github.com/mozilla-mobile/android-components/issues/4704
